minimal_high_density.md

```diff
@@ -0,0 +1,103 @@
+# `minimal_high_density.slim`: a minimal model suitable for high-density simulations
+
+**code:** [minimal_high_density.slim](minimal_high_density.slim); on [github](https://github.com/kr-colab/spatial_sims_standard/blob/main/minimal_high_density.slim)
+
+This code implements the same model as [`minimal.slim`](minimal.html),
+but implements some "map-based" alternative computational methods so that runtime is linear in density
+instead of quadratic.
+However, these methods are somewhat approximate and may lead to discretization artifacts if density is low,
+so if neighborhood sizes are small (below, say 5 or 10), then the methods should probably not be used.
+
+There are two aspects of the "standard" methods demonstrated in `minimal.slim` that are quadratic in population density
+(i.e., quadratic in the parameter `K`): local density computation, and mate choice.
+The two methods can be used independently - for instance, maybe your simulation
+has small interaction neighborhood size but large mating neighborhood size,
+so you'd use these methods for mating but not density computations.
+
+Both methods are linear because they rely on a pre-computed map of local density.
+The basic computation to do this is
+```
+raw = summarizeIndividuals(p1.individuals, GRID_DIMS, p1.spatialBounds, operation="individuals.size();", perUnitArea=T);
+```
+This makes a grid of dimensions `GRID_DIMS`, counts up the number of individuals in each grid square,
+and divides by the area of the square to get a density per unit area.
+What is an appropriate size for these grid squares?
+Well, we'd like these to be as big as is reasonable,
+because computation will scale with the number of these squares.
+But, this grid is the source of any discretization artifacts:
+the smaller the grid squares are, the closer this model is to `minimal.slim`.
```

Comment:
or even worse, potentially!  could be *much* worse if you make them really small!